lib/DDGC/Util/Script/Docs.pm

```diff
@@ -0,0 +1,283 @@
+package DDGC::Util::Script::Docs;
+
+use Moo;
+
+use File::Find;
+use File::Spec;
+use File::chdir;
+use File::Temp qw/ tempdir /;
+use File::Copy qw/ mv /;
+use Path::Class;
+use IO::All;
+use Markdent::Handler::HTMLStream::Document;
+use Markdent::Parser;
+use Text::Trim;
+use JSON;
+use Text::Xslate;
+use Try::Tiny;
+use IPC::Run qw{run timeout};
+
+with qw/
+    DDGC::Util::Script::Base::Service
+    DDGC::Util::Script::Base::ServiceEmail
+/;
+
+has xslate => (
+    is => 'ro',
+    lazy => 1,
+    builder => sub {
+        Text::Xslate->new(
+            path => 'views/duckduckgo-documentation',
+            verbose => 2,
+        );
+    },
+);
+
+has docs_dir => (
+    is => 'ro',
+    lazy => 1,
+    builder => sub { $_[0]->ddgc_config->ddh_docs_directory; },
+);
+
+has nav_file => (
+    is => 'ro',
+    lazy => 1,
+    builder => sub {'ddh-index.md'},
+);
+
+has prev_next_file => (
+    is => 'ro',
+    lazy => 1,
+    builder => sub {'ddh-prev-next.json'},
+);
+
+has source_dir => (
+    is => 'ro',
+    lazy => 1,
+    builder => '_build_source_dir',
+);
+sub _build_source_dir {
+    my ( $self ) = @_;
+
+    my $cache_dir = $self->ddgc_config->cachedir;
+    my $source_dir = File::Spec->catdir(
+        $cache_dir, 'duckduckgo-documentation/duckduckhack'
+    );
+
+    {
```

Comment:
Not entirely sure - this is a c/p job from publisher (refactoring that code is out-of-scope for this, I think). I'm guessing it's because we use [File::chdir](https://metacpan.org/pod/File::chdir) within the block, but `$CWD` changes appear to be scoped within the `if` block anyway, so ¯\_(ツ)_/¯